src/wp-includes/rest-api/endpoints/class-wp-rest-template-revisions-controller.php

```diff
@@ -160,18 +160,23 @@ public function register_routes() {
 	 *
 	 * @since 6.4.0
 	 *
-	 * @param int $parent_post_id Supplied ID.
+	 * @param string $parent_post_id Supplied ID.
 	 * @return WP_Post|WP_Error Post object if ID is valid, WP_Error otherwise.
 	 */
 	protected function get_parent( $parent_post_id ) {
+		$error = new WP_Error(
```

Comment:
The `$error` is instantiated even when the parent post ID is not null and the template exists. It would make sense to me to create a separate private function `build_invalid_parent_error` and do something like:
```
if ( null === $parent_post_id ) {
	return $this-build_invalid_parent_error();
}

$template = get_block_template( $parent_template_id, $this->parent_post_type );
if ( ! $template ) {
	return $this-build_invalid_parent_error();
}

return get_post( $template->wp_id );
```

Reply:
After reading the whole discussion I actually agree with @TimothyBJacobs. Let's close this abs the linked ticket.

On a separate note, I don't like the error instantiation when it's not needed. I am going to create a ticket to address that shortly.

Reply:
@martinkrcho I have closed the PR, I don't think I can close the ticket?